tests/extension/oneapi_device_global/device_global_exceptions.cpp

```diff
@@ -0,0 +1,289 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Provides tests for the exceptions that are thrown by sycl::queue and
+//  sycl::handler copy methods overloads
+//
+*******************************************************************************/
+
+#include "../../common/common.h"
+#include "../../common/get_cts_object.h"
+#include "device_global_common.h"
+#include "type_pack.h"
+
+#define TEST_NAME device_global_exceptions
+
+namespace TEST_NAMESPACE {
+using namespace sycl_cts;
+using namespace sycl_cts::util;
+using namespace device_global_common_functions;
+
+#if defined(SYCL_EXT_ONEAPI_PROPERTY_LIST) && \
+    defined(SYCL_EXT_ONEAPI_DEVICE_GLOBAL)
+namespace oneapi = sycl::ext::oneapi;
+
+/**
+ * @brief The functions checks if error code of the given exception is equal to
+ * sycl::errc::invalid
+ * @param e The sycl::exception instance that needs to verify error code
+ * @return True if error code of exception is sycl::errc::invalid, false
+ * otherwise
+ */
+inline bool is_errc_invalid(const sycl::exception& e) {
+  return e.code() != sycl::errc::invalid;
+}
+
+namespace handler_copy_exceptions {
+// Creating instance with default constructor
+template <typename T>
+oneapi::device_global<T> dev_global;
+
+/** @brief The function tests that .copy() method overload throws an exception
+ * with error code equal to errc::invalid if attempt to write beyond the end of
+ * the destination variable
+ *  @tparam T Type of underlying device_global value
+ */
+template <typename T, size_t N>
+void run_test(util::logger& log, const std::string& type_name) {
+  std::shared_ptr<T[N]> src(new T[N]);
+  std::shared_ptr<T[N]> dest(new T[N]);
+
+  bool is_exception_thrown{true};
+  bool is_exception_correct{true};
+  auto queue = util::get_cts_object::queue();
+
+  queue.submit([&](sycl::handler& cgh) {
+    // Try to write data beyond the end of the destination variable
+    // Have to throw an exception
+    try {
+      cgh.template copy<T>(src.get(), dev_global<T>, N, N / 2);
+      is_exception_thrown &= false;
+    } catch (sycl::exception const& e) {
+      is_exception_correct &= is_errc_invalid(e);
+    }
+
+    try {
+      cgh.template copy<T>(dev_global<T>, dest.get(), N, N / 2);
+      is_exception_thrown &= false;
+    } catch (sycl::exception const& e) {
+      is_exception_correct &= is_errc_invalid(e);
+    }
+  });
+  queue.wait_and_throw();
+
+  if (!is_exception_thrown) {
+    FAIL(log, get_case_description(
+                  "deivice_global: sycl::handler .copy() method exception",
```

Comment:
Change everywhere
```suggestion
                  "device_global: sycl::handler .copy() member function exception",
```

Reply:
Applied change to every comment with "method"

Reply:
You need also to fix the typo `deivice_global` → `device_global` everywhere.

Reply:
Thank you, I missed the other ones. Fixed in this PR.